### DISCUSSION TITLE
Should SDNext and Automatic1111 produce the same image when using the same settings?

### DISCUSSION THREAD

**Discussion by OuticNZ** [vladmandic/sdnext#2840] [General]
Should I be able to reproduce an image that was created in Automatic1111 in SD.Next?  

Is it something I'm doing wrong?  Is that not how it's supposed to work?

Even copying the prompt from say civitai.com for an image and pasting into SD.Next - it doesn't create the same image.  (Pretty sure).  Will verify this.  

Background:
I'm running Unraid, and have deployed Automatic1111 and SD.Next into two different docker containers.  Both are new installs - using the latest available (main) release - not development.  

Below is an image created in Automatic1111.
![00008-984186962](https://github.com/vladmandic/automatic/assets/30253830/cade6123-6ae6-4698-bc47-b7d919ddb87e)
```
Prompt: concept art, a cat, driving a car, cyberpunk

Negative: 

Parameters: Steps: 20| Sampler: DPM++ 2M Karras| CFG scale: 7| Seed: 984186962| Size-1: 512| Size-2: 512| Model hash: 48de823bb4| Model: 0001softrealistic_v155xxx| VAE hash: c6a580b13a| VAE: vae-ft-mse-840000-ema-pruned.ckpt| Version: v1.7.0
```

If I take this image in Automatic1111 and send the PNG Info to Text to Image.  I get the same image.  If I do this in SDNext I get a completely different image.
![00009-0001softrealistic_v155xxx-concept art a cat driving a car cyberpunk](https://github.com/vladmandic/automatic/assets/30253830/5c285c49-bffe-43b8-924b-896aba7ee200)
```
Prompt: concept art, a cat, driving a car, cyberpunk

Negative: 

Parameters: Steps: 20| Seed: 984186962| Sampler: UniPC| CFG scale: 7| Size-1: 512| Size-2: 512| Parser: Full parser| Model: 0001softrealistic_v155xxx| Model hash: 48de823bb4| Backend: Original| App: SD.Next| Version: 3c95267| Operations: txt2img| Sampler brownian: False| Sampler sigma: karras
```

The sampler isn't being set correctly for some reason in SD.Next.  It's using UniPC even though the UI is set to DPM++ 2M Karras.
![image](https://github.com/vladmandic/automatic/assets/30253830/2356ad9e-a27c-47a0-9bf0-6ed1cbb3d6f2)

If I change it to another sampler and then back to the correct one it then uses the correct one, but doesn't change the outcome.
![00010-0001softrealistic_v155xxx-concept art a cat driving a car cyberpunk](https://github.com/vladmandic/automatic/assets/30253830/dedc1801-c711-4821-a68d-59bdc0eccbd1)
```
Prompt: concept art, a cat, driving a car, cyberpunk

Negative: 

Parameters: Steps: 20| Seed: 984186962| Sampler: DPM++ 2M| CFG scale: 7| Size-1: 512| Size-2: 512| Parser: Full parser| Model: 0001softrealistic_v155xxx| Model hash: 48de823bb4| Backend: Original| App: SD.Next| Version: 3c95267| Operations: txt2img| Sampler brownian: False| Sampler sigma: karras

```

**Comment by vladmandic** [comment]
sdnext using backend original will produce near-identical images for same settings.
sdnext using backend diffusers is a different story. its like comparing a1111 with invokeai or comfy - no, they will not produce identical images.

**Reply by OuticNZ** [reply]
In this example SDNext is using the original backend in this case, so I would expect near-identical images, not wildly different.
![image](https://github.com/vladmandic/automatic/assets/30253830/0c649312-3234-48d3-bb8d-0dbeddb73e8a)

**Comment by vladmandic** [comment]
Post a link to model, I'll check it out.

**Reply by OuticNZ** [reply]
This is the model link @ CivitAI - [0001softrealistic](https://civitai.com/models/85731/0001softrealistic) v155

I can't work it out.  A fresh install of Automatic1111, and it works fine, a fresh install of SD.Next and setting to original backend, gets different results.  Really trying to understand why and how to fix.

Will try with other models.

**Reply by OuticNZ** [reply]
So I have tested a different model and I'm getting the same image created.

[Photographer Alpha v3](https://civitai.com/models/179674/photographer-alpha7)  

Although I have noticed an issue when sending the image to Text.

Reference Image created on Automatic1111:
![00001-984186962](https://github.com/vladmandic/automatic/assets/30253830/c021f2c0-0100-411c-9f96-75b009d7f4a7)
```
concept art, a cat, driving a car, cyberpunk
Steps: 20, Sampler: DPM++ 2M Karras, CFG scale: 7, Seed: 984186962, Size: 512x512, Model hash: 64d899ae35, Model: photographerAlpha7_v3, Version: v1.7.0
```

SD.Next
1. Click on **Process** and drag reference image onto Source
![image](https://github.com/vladmandic/automatic/assets/30253830/3007da4a-be78-4b02-97d9-7e995f982379)
2.Note properties from file
![image](https://github.com/vladmandic/automatic/assets/30253830/b0b52fe6-1366-4c9b-982f-06102fca37e4)
3. Click **->Text**
4. Note Sampling method is set to DPM++ 2M Karras.  This is correct for Automatic1111, but in SD.Next its not a valid option
![image](https://github.com/vladmandic/automatic/assets/30253830/284aa827-79ba-49ca-99f9-bc61ef91dca9)
5. Click **Generate**
![image](https://github.com/vladmandic/automatic/assets/30253830/ef1d67e6-bf7c-4f86-9c00-b492da99b3f7)
6. Notice Sampler is UniPC and not DPM++ 2M Karras
7. Manually set Sampler to **DPM++ 2M** and set Sigma Algortihm to **karras**
![image](https://github.com/vladmandic/automatic/assets/30253830/38c4e196-fdc6-42d3-8bb4-97684224bdba)
8. Click **Generate**
![image](https://github.com/vladmandic/automatic/assets/30253830/c981e19e-5739-4109-a389-76e0289bb47b)

Same image.  This isn't working for the other model which I find odd, but hey maybe that is normal.  

In Automatic1111 the same image can be generated regardless of the model used.  For some reason in SD.Next there's a difference for some models.  Again this needs more work to find out just what models are consistent and which are not.

**Reply by vladmandic** [reply]
this is helpful.
and regarding samplers, yes, they have different names between sdnext and a1111, so settings cannot be just "applied" without setting them manually.

**Comment by OuticNZ** [comment]
Some more testing
Model: [Photon v1](https://civitai.com/models/84728/photon?modelVersionId=90072) - reference image:  https://civitai.com/images/1044343
    Automatic1111 - Produces the same image
    SD.Next - Produces the same image

Model:  [GhostMix](https://civitai.com/models/36520/ghostmix?modelVersionId=76907) - reference image:  https://civitai.com/images/862136
![image](https://github.com/vladmandic/automatic/assets/30253830/8676df91-6e6e-4f67-b6d7-174aea4ca886)

    Automatic1111 - Produces a near replica - there are some differences.
![00009-1015468391](https://github.com/vladmandic/automatic/assets/30253830/ede18e62-08a7-4b69-8fbc-cb805c8e9b64)

    SD.Next - A different image and colours more vibrant.  This did have the setting eta_noise_seed_delta set, but SD.Next doesn't seem to have it.  
![00068-ghostmix_v20Bakedvae-masterpiece top quality best quality official art beautiful](https://github.com/vladmandic/automatic/assets/30253830/1e5e9b58-b0fc-4801-88ac-a3690160fe36)

**Comment by OuticNZ** [comment]
So some more testing results.   Using several checkpoint models and a simple prompt.  I have tested 3 setups.

1) Automatic1111
2) SDNext (using Original backend)
3) SDNext (using Diffusers)

Models:

- aZovyaPhotoreal v2
- absolutereality v1.81
- analog Madness v7
- cyberrealistic v4.1
- classic by stable yogi v2
- epic realism pure evolution v5
- human realistic v10
- object reality v2
- natural reality v2
- photographer Alpha7 v3
- photon v1
- realistic vision v6 B1 VAE
- the truality engine v3
- realism by stable yogi v3

The results, no detectable difference in output image.   Will work on the next set of tests. 
![results01-smaller](https://github.com/vladmandic/automatic/assets/30253830/5acef3df-0736-4f01-bf12-d4fb3cf5e704)
